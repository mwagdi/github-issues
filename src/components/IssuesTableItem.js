import React from 'react';
import { Link } from 'react-router-dom';

import OpenSVG from './svgs/OpenClosed';
import { getTimeAgo } from '../helpers';

const IssuesTableItem = ({issue}) => (
    <div className="issues-table__item flex-container">
        <div className="flex-grow flex-container">
            <div className="flex-container flex-column">
                <div className="flex-container issues-table__item_name">
                    <OpenSVG open={issue.state === "open"} />
                    <Link
                    to={`/${issue.number}`}
                    className="issues-table__link">
                        {issue.title}
                    </Link>
                </div>
                <span className="opened">
                    {`#${issue.number} opened ${getTimeAgo(issue.created_at)} by ${issue.user.login}`}
                </span>
            </div>
        </div>
        <Link
        to={`/${issue.number}`}
        className="flex-container">
            <svg className="octicon octicon-comment v-align-middle comments-icon" viewBox="0 0 16 16" version="1.1" width="16" height="16" aria-hidden="true">
                <path fillRule="evenodd" d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"></path>
            </svg>
            <span className="comments-number">{issue.comments}</span>
        </Link>
    </div>
);

export default IssuesTableItem;