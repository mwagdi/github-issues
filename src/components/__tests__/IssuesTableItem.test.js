import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import IssuesTableItem from '../IssuesTableItem';


test('IssuesTableItem renders correctly', () => {
    const issue = {
        user: {
            login: "johndoe"
        },
        created_at: "10/10/2017"
    }
    const wrapper = shallow(<IssuesTableItem issue={issue} />);
    expect(toJSON(wrapper)).toMatchSnapshot();
})
