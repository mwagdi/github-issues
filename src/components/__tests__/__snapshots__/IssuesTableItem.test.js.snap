// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssuesTableItem renders correctly 1`] = `
<div
  className="issues-table__item flex-container"
>
  <div
    className="flex-grow flex-container"
  >
    <div
      className="flex-container flex-column"
    >
      <div
        className="flex-container issues-table__item_name"
      >
        <OpenClosed
          open={false}
        />
        <Link
          className="issues-table__link"
          replace={false}
          to="/undefined"
        />
      </div>
      <span
        className="opened"
      >
        #undefined opened on Oct 10, 2017 by johndoe
      </span>
    </div>
  </div>
  <Link
    className="flex-container"
    replace={false}
    to="/undefined"
  >
    <svg
      aria-hidden="true"
      className="octicon octicon-comment v-align-middle comments-icon"
      height="16"
      version="1.1"
      viewBox="0 0 16 16"
      width="16"
    >
      <path
        d="M14 1H2c-.55 0-1 .45-1 1v8c0 .55.45 1 1 1h2v3.5L7.5 11H14c.55 0 1-.45 1-1V2c0-.55-.45-1-1-1zm0 9H7l-2 2v-2H2V2h12v8z"
        fillRule="evenodd"
      />
    </svg>
    <span
      className="comments-number"
    />
  </Link>
</div>
`;
